AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/Mutations.dfy

```diff
@@ -90,6 +95,28 @@ module {:options "/functionSyntax:4" } Mutations {
     var kmsOperation: string;
     var success?: bool := false;
     var throwAwayError;
+    // TODO-HV-2-M3: Can this if condition also handle hv-2 item?
+    if (isTerminalHv2?) {
```

Comment:
This condition is too broad;
either we semantically need `isTerminalHv2? && item.hierarchyVersion.v2?`,
or we need to create a new method that 
calls KMS Decrypt for either HV-1 or HV-2 
appropriately.

We can probably refactor `GetKeys.dfy` 
to have a generic `UnwrapBranchKeyItem` or 
`DecryptAndValidateBranchKeyItem` method
that all three of the `Get*` operations can call.

In fact, the predicate you wrote,
`ValidateKmsDecryption`,
is tantalizingly close to a generic BK Item un-wrapper,
that could collapse a significant amount of the logic in `GetKeys.dfy`
and make this logic simpler...